src/js/components/Box/StyledBox.js

```diff
@@ -191,14 +191,22 @@ const interactiveStyle = css`
   cursor: pointer;
 
   &:hover {
+    ${(props) =>
+      props.kind?.hover &&
+      getHoverIndicatorStyle(props.kind.hover, props.theme)}
     ${(props) =>
       props.hoverIndicator &&
       getHoverIndicatorStyle(props.hoverIndicator, props.theme)}
   }
 `;
 
 // NOTE: basis must be after flex! Otherwise, flex overrides basis
-const StyledBox = styled.div`
+const StyledBox = styled.div.withConfig({
```

Comment:
Can we just rename `kind` to `kindProp` to prevent it leaking? It feels like we do this elsewhere.

Reply:
I will give this a try